### PR TITLE
Chore: APP-2050 - React Query Setup

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description

- adds initial configuration for react-query
- refactors `useDaos` hook -> `useDaosQuery`

Task: [APP-2050](https://aragonassociation.atlassian.net/browse/APP-2050)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-2050]: https://aragonassociation.atlassian.net/browse/APP-2050?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/web-app/src/containers/daoExplorer/daoExplorer.tsx [hunk 1]
```diff
@@ -114,46 +87,39 @@ export const DaoExplorer = () => {
...
-            displayedDaos.map((dao, index) => (
+            infiniteDaos?.pages?.map(dao => (
               <DaoCard
+                key={dao.address}
                 name={dao.metadata.name}
                 ensName={dao.ensDomain}
                 logo={dao.metadata.avatar}
                 description={dao.metadata.description}
-                chainId={dao.chain || CHAIN_METADATA[network].id} // Default to Goerli
+                chainId={dao.chain}
```
Comment (sepehr2github) [inline packages/web-app/src/containers/daoExplorer/daoExplorer.tsx:127]: Quick question, why did you removed CHAIN_METADATA[network].id, the dao.chain will be supported by the new SDK?

File: packages/web-app/src/services/cache.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * @returns list of favorited DAOs based on given options
+ */
+export async function getFavoritedDaosFromCache(
+  cache: Array<NavigationDao>,
+  options: {skip: number; limit: number}
+): Promise<NavigationDao[]> {
+  const {skip, limit} = options;
+
+  // sleeping for 600 ms because the immediate apparition of DAOS creates a flickering issue
+  await sleepFor(600);
```
Comment (sepehr2github) [inline packages/web-app/src/services/cache.ts:21]: 600 is just a random number right?

**Review by sepehr2github — APPROVED** [review]
Technically It works fine and I see no issue but There are still a couple of question from my side that are worth asking, Thanks a lot, dear fabric for bringing the fantastic react query to our project

File: packages/web-app/src/containers/daoExplorer/daoExplorer.tsx (hunk 1 above)
Comment (Fabricevladimir) [reply packages/web-app/src/containers/daoExplorer/daoExplorer.tsx:127]: That is the long term plan yes. Currently in the `useDaosQuery.tsx`, there is a transformation function (`toAugmentedDaoListItem`) that adds the proper `chain` to the DAOs that are returned by the SDK. The idea is that once the SDK returns the chain id on the `DaoListItem` type, we can remove the code that does the augmentation in the app.

File: packages/web-app/src/services/cache.ts (hunk 2 above)
Comment (Fabricevladimir) [reply packages/web-app/src/services/cache.ts:21]: Essentially, yes. I'm certainly open to suggestions if the data is returned to quickly or not quickly enough.

**Pull request merged**